commands/operator-sdk/cmd/generate/openapi.go

```diff
@@ -111,26 +109,29 @@ func OpenAPIGen() error {
 		AbsProjectPath: absProjectPath,
 		ProjectName:    filepath.Base(absProjectPath),
 	}
-	crdMap, err := getCRDGVKMap()
+	crds, err := k8sutil.GetCRDs(scaffold.CRDsDir)
 	if err != nil {
 		return err
 	}
-	for g, vs := range gvMap {
-		for _, v := range vs {
-			gvks := crdMap[filepath.Join(g, v)]
-			for _, gvk := range gvks {
-				r, err := scaffold.NewResource(filepath.Join(gvk.Group, gvk.Version), gvk.Kind)
-				if err != nil {
-					return err
-				}
-				err = s.Execute(cfg,
-					&scaffold.CRD{Resource: r, IsOperatorGo: projutil.IsOperatorGo()},
-				)
-				if err != nil {
-					return err
-				}
+	for _, crd := range crds {
+		g, v, k := crd.Spec.Group, crd.Spec.Version, crd.Spec.Names.Kind
+		if v == "" {
+			if len(crd.Spec.Versions) != 0 {
+				v = crd.Spec.Versions[0].Name
```

Comment:
@estroz This reminds me that we don't have validation for multiple CRD versions fully figured out yet.
We have a [top level validation block](https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/types.go#L46-L50) and the [per version validation block](https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/types.go#L46-L50) which seems to be alpha right now.

If we allow users to define multiple versions in a single CRD file e.g `cache_v1alpha_memcahced_crd.yaml`:
```YAML
spec:
  group: cache.example.com
  versions:
  - name: v1beta1
    served: true
    storage: true
  - name: v1alpha1
    served: true
    storage: false
```
then we'll end up generating the top level validation block for v1beta1:
```YAML
spec:
  group: cache.example.com
  versions:
  - name: v1beta1
    served: true
    storage: true
  - name: v1alpha1
    served: true
    storage: false
  validation:
  ...
```
instead of the per version validation blocks:
```YAML
spec:
  group: cache.example.com
  versions:
  - name: v1beta1
    served: true
    storage: true
    validation: 
        ...
  - name: v1alpha1
    served: true
    storage: false
    validation: 
        ...
```

I don't want to figure it out in this PR but thought I'd point it out since we're starting to deal with multiple crd versions now.
Can you check what `generate openapi` does when you have multiple versions defined in the `versions` field with this change?

Reply:
As of now `generate openapi` doesn't handle multiple versions because it uses `crd.Spec.Version`. This PR allows us to generate validation blocks for multiple versions in the future. We'll have to think through how/if we want to change CRD manifest generation to support multiple versions.